### PR TITLE
Add Styles API example for vanilla leaflet

### PR DESCRIPTION
**Pull request by danswick**
An example demonstrating how to use Styles API tiles by setting tile size and zoom offset in Leaflet. 

cc: @bsudekum @lyzidiamond

### CODE REVIEW

**Comment by lyzidiamond** [comment]
cc @jfirebaugh @lucaswoj

File: Gemfile.lock [hunk 1]
```diff
@@ -75,4 +75,4 @@ DEPENDENCIES
   jekyll (= 2.1.1)
   liquid (= 2.6.1)
   rack-jekyll
-  rake
+  rake
```
Comment (lucaswoj) [inline Gemfile.lock]: Please revert this

File: docs/_posts/examples/v1.0.0/0100-01-01-leaflet-studio-style.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  - leaflet
+---
+<div id='map'></div>
+
+<script>
+var map = L.map('map').setView([38.97416, -95.23252], 15);
+
+// Add tiles from Mapbox Style API (https://www.mapbox.com/developers/api/styles/)
+// Tiles are 512x512 pixels and are offset by 1 zoom level
+L.tileLayer('https://api.mapbox.com/styles/v1/mapbox/emerald-v8/tiles/{z}/{x}/{y}@2x?access_token=' + L.mapbox.accessToken, {tileSize: 512, zoomOffset: -1}).addTo(map);
```
Comment (lucaswoj) [inline docs/_posts/examples/v1.0.0/0100-01-01-leaflet-studio-style.html]: This line is a bit too long. I'd write it like this (different folks have different styles)

``` js
L.tileLayer(
  'https://api.mapbox.com/styles/v1/mapbox/emerald-v8/tiles/{z}/{x}/{y}@2x?access_token=' + L.mapbox.accessToken,    
  {tileSize: 512, zoomOffset: -1}
).addTo(map);
```

**Comment by jfirebaugh** [comment]
- `@2x` should be omitted from the URL
- Should add an attribution like in https://www.mapbox.com/mapbox.js/example/v1.0.0/plain-leaflet/

**Comment by danswick** [comment]
Thanks @jfirebaugh and @lucaswoj. Edits made. Apologies for the Git flailing in that last set of commits.

**Comment by lucaswoj** [comment]
Looks good to me. Squash into one commit and :shipit:.

**Comment by jfirebaugh** [comment]
Sorry for not noticing this earlier, but the attribution format is out of date (here and in https://www.mapbox.com/mapbox.js/example/v1.0.0/plain-leaflet/). Would you mind updating both examples to follow [this format from our help docs](https://www.mapbox.com/help/attribution/#other-mapping-frameworks)?

**Pull request merged**